### DISCUSSION TITLE
How to write a view function which could return different types of Responses(like either an empty 400 or a redirect)?

### DISCUSSION THREAD

**Discussion by farooqkz** [tokio-rs/axum#2015] [Q&A] [answered]
Hello. I have a function which depending on some conditions, will return either a StatusCode with an empty body or a Redirect. I don't know what "type" should the function return.

Edit: Similarly, in some other function, I have to return either a 400 with empty body or a 200 with json body. I am pretty new to Rust. Perhaps Rust has some "type" which is either "A type" or "B type"?

**Comment by link2xt** [comment, marked as answer]
The function should return `Response`. See an example here of how to convert both types of responses to a common type: https://docs.rs/axum/0.6.18/axum/response/index.html#returning-different-response-types

**Reply by farooqkz** [reply]
So the key is `.into_response()`. Thank you!
